app/models/recipe.rb

```diff
@@ -1,63 +1,40 @@
 require 'net/http'
 class Recipe < ActiveRecord::Base
     has_and_belongs_to_many :users
-    def self.restructure_hash(res_json_hash, all_ids)
```

Comment:
The refactoring of this method looks really good! Makes everything far cleaner and more concise. Well done!